src/libtexture/imagecache.cpp

```diff
@@ -2591,33 +2615,49 @@ ImageCacheImpl::get_image_info(ImageCacheFile* file,
         *(int*)data = file->is_udim();
         return true;
     }
-    if (file->is_udim() && dataname == s_channels) {
-        // Special case -- it's ok to ask for a UDIM's channels. It'll
-        // search for a concrete file. Beware, this will cause trouble
-        // if different panels of the same UDIM scheme have different
-        // numbers of channels! Search the 10x100 files for a match, give
-        // up if not found.
-        for (int j = 0; j < 100; ++j) {
-            for (int i = 0; i < 10; ++i) {
-                float s = i + 0.5f, t = j + 0.5f;
-                ImageCacheFile* concretefile = resolve_udim(file, thread_info,
-                                                            s, t);
-                concretefile = verify_file(concretefile, thread_info, true);
-                if (concretefile && !concretefile->broken()) {
-                    // Recurse to try again with the concrete file
-                    bool r = get_image_info(concretefile, thread_info, subimage,
-                                            miplevel, dataname, datatype, data);
-                    // suppress errors from missing tiles we encountered
-                    // prior to finding the one that worked.
-                    (void)geterror();
-                    return r;
+
+    if (file->is_udim()) {
+        // UDIMs are tricky. What we want to do here is successfully
+        // retrieve the attribute if it's the same for all UDIM tiles, but
+        // fail if it's not the same for all tiles.
```

Comment:
The one place where I strongly disagree with this for is resolution.
Resolution can be, and often is, very different for different UDIMs. So if you can query resolution differently for each UDIM, then that's probably fine. But that restriction seems unneccessarily restrictive. At least for how we work at AL.

Reply:
You can certainly have different UDIM file each with different resolution. You just can't ask for the resolution (if they differ) in this get_image_info() call that doesn't know at u,v panel you want to know about. Maybe we need a get_image_info variety that also takes texture coordinates?